extension/src/popups/MinervaMenu.js

```diff
@@ -21,21 +23,27 @@ function displayClassRankings() {
     });
 }
 
-window.onload = function() {
-    chrome.storage.local.get(["student_name"], (result) => {
-        if (result.hasOwnProperty("student_name")) {
-            transitionToMainMenu(result["student_name"]);
+function diplayCurrentStudentInfo(studentName) {
+    chrome.runtime.sendMessage({
+        type: "FetchCurrentStudentInfo",
+        studentName: studentName
+    }, (response) => {
+        if (response.status) {
+            renderCurrentStudentInfo(response);
```

Comment:
I am not gonna make you change this, but it would help you a lot in terms of code callchain if you set/get global variables

Reply:
That would be better.